src/api/api.ts

```diff
@@ -1,120 +1,217 @@
+      console.error('Error fetching data:', error);
+      toggleAlertError('Error occured. Please contact with administrator.');
     }
 }
 
-export const fetchUserData = async (userId:string) => {
-    const { data, error } = await supabase
-    .from('income')
-    .select('*')
-    .eq('id', userId)
-    if (error) throw error;
-    if (data) {
-        return data
+export const fetchUserData = async (userId: string, toggleAlertError: ((arg0: string) => void)) => {
+    try {
+      const response = await fetch(`${PATH}/income/${userId}`);
+      if (!response.ok) {
+        throw new Error('Network response was not ok');
+      }
+  
+      const data = await response.json();
+      return data
+    } catch (error) {
+      console.error('Error fetching data:', error);
+      toggleAlertError('Error occured. Please contact with administrator.');
     }
 }
 
 export const addMonth = async (values:AddMonthData, userId: string, toggleAlertSuccess: { (alert: string): void; (arg0: string): void; }, toggleAlertError: { (alert: string): void; (arg0: string): void; }) => {
     const dbData: (string | string[])[] = [];
     const valuesData = values.month.concat((values.year).toString());
-    const { data, error } = await supabase
-    .from('income')
-    .select('incomeId, monthName, year')
-    if (error) throw error;
-    data.map(async (el: { monthName: string; year: number; })=>{
-        dbData.push((el.monthName).concat(el.year.toString()));
-    })
-    const check = dbData.includes(valuesData);
-    if (check) {
-        toggleAlertError('Month already exist!');
-        return
-    } else {
-        const { data:data2, error:error2 } = await supabase
-            .from('income')
-            .insert([
-              { id: userId, incomeId: data.length+1, monthIncome: values.income, monthName: values.month, year: values.year }
-            ])
-            if (error2) throw error2;
-            toggleAlertSuccess('Month added!');
-            return data2;
-    }
+    try {
+        const response = await fetch(`${PATH}/getMonths`);
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+    
+        const data = await response.json();
+        data.map(async (el: { monthName: string; year: number; })=>{
+            dbData.push((el.monthName).concat(el.year.toString()));
+        })
+        const check = dbData.includes(valuesData);
+        if (check) {
+            toggleAlertError('Month already exist!');
+            return
+        } else {
+            try {
+                const response2 = await fetch(`${PATH}/addMonth`, {
+                  method: 'POST',
+                  headers: {
+                    'Content-Type': 'application/json',
+                  },
+                  body: JSON.stringify({ userId, data, values }),
+                });
+            
+                if (!response2.ok) {
+                  throw new Error('Network response was not ok');
```

Comment:
Let's discuss these error codes next time we meet :) I think they can be extended a little bit